### DISCUSSION TITLE
Thumbnail extraction for RAW photos

### DISCUSSION THREAD

**Discussion by luckydye** [spacedriveapp/spacedrive#1161] [Feature Request]
Hi, im wondering if there is already any work done for RAW Image thumbnails? 

I have a basic but working implementation of some formats for Canon, Nikon and Sony files in my Fork which I would love to contribute.

**Comment by utkubakir** [comment]
We haven't done any work but it's in our to-do list. I think we will start working on it this week or the next.

**Reply by luckydye** [reply]
Ah cool. Here is my branch if someone wants to check out how I did it: https://github.com/luckydye/spacedrive/tree/sd-images

**Reply by brxken128** [reply]
We already have some work done on this somewhere I believe, as well as for RAW images.

I'm not too huge on how they're converted to JPEG first, unless I'm misreading the code in which case I apologise. I think we can optimise it further.

**Reply by luckydye** [reply]
I couldnt find anything related to raw images. 
Most vendors embed JPEG previews in their raw files which I find and extract. I thought that would be the better performing thing to do than converting the actual raw data. But I see how that can sometimes be not the ideal way.

**Reply by brxken128** [reply]
Yeah, we do our own thumbnail processing on the raw image data ourselves otherwise one thumbnail could be 4k while the other 360p. This way, we're able to keep all thumbnail sizes consistent.
